packages/core/useFocus/index.ts

```diff
@@ -34,34 +33,23 @@ export interface FocusReturn {
 export function useFocus(options: FocusOptions = {}): FocusReturn {
   const {
     initialValue = false,
-    window = defaultWindow,
   } = options
 
-  const focused = ref(initialValue)
-
-  if (!window) return { focused }
-
-  const onFocus = () => { focused.value = true }
-  const onBlur = () => { focused.value = false }
-
-  const target = computed(() => unrefElement(options.target) ?? window)
-
-  useEventListener(target, 'focus', onFocus, { passive: true })
-  useEventListener(target, 'blur', onBlur, { passive: true })
-
-  const setFocus = (focused: boolean, oldFocused: boolean | undefined) => {
-    if (focused) {
-      if (!oldFocused) target.value?.focus()
-    }
-    else {
-      if (oldFocused) target.value?.blur()
-    }
-  }
-
-  watch(focused, setFocus, { immediate: true, flush: 'post' })
-  watch(target, () => {
-    setFocus(focused.value, false)
-  }, { immediate: true, flush: 'post' })
+  const activeElement = useActiveElement()
+  const target = computed(() => unrefElement(options.target))
+  const focused = computed({
+    get() {
+      return activeElement.value === target.value
+    },
+    set(value: boolean) {
+      if (!value && focused.value)
+        target.value?.blur()
+      if (value && !focused.value)
+        target.value?.focus()
+    },
+  })
+
+  watch(target, () => { focused.value = initialValue }, { immediate: true, flush: 'post' })
```

Comment:
If the `intialValue` is set to true then this will result in the element being focused each time the target ref is updated. So for conditionally rendered elements with `v-if` it will cause them to be focused each time they render (not just the first time).

So for my example in #926 it will focus the input each time the show flag is toggled to true.

Not sure if this is desired or what the expected behavior should be.